src/Routers/FilesRouter.js

```diff
@@ -5,6 +5,31 @@ import Parse from 'parse/node';
 import Config from '../Config';
 import mime from 'mime';
 import logger from '../logger';
+const triggers = require('../triggers');
+const http = require('http');
+
+const downloadFileFromURI = (uri) => {
+  return new Promise((res, rej) => {
+    http.get(uri, (response) => {
+      response.setDefaultEncoding('base64');
+      let body = `data:${response.headers['content-type']};base64,`;
+      response.on('data', data => body += data);
+      response.on('end', () => res(body));
+    }).on('error', (e) => {
+      rej(`Error downloading file from ${uri}: ${e.message}`);
+    });
+  });
+};
+
+const addFileDataIfNeeded = async (file) => {
+  if (file._source.format === 'uri') {
+    const base64 = await downloadFileFromURI(file._source.uri);
+    file._previousSave = file;
+    file._data = base64;
+    file._requestTask = null;
+  }
+  return file;
+};
```

Comment:
I don't think we should be downloading on our end. There is `file.getData()` that the user could call that exist for downloading. We can update it to also get file size etc.

I was planning on creating a new endpoint to get file data. Mongo for example would return something like this.

```
[ { _id: 5e779be8f4bee4eb4b8ea6fd,
    length: 9,
    chunkSize: 261120,
    uploadDate: 2020-03-22T17:10:01.012Z,
    filename: 'myFileName',
    md5: '8aa99b1f439ff71293e95357bac6fd94',
    metadata: { hello: 'world' } } ]
```

This can be done in a separate PR.

Reply:
The `file.getData()` function is only for downloading the data for files that have already been saved. What i'm doing here is accounting for a `Parse.File` that is returned in the beforeSaveFile hook that was created using `new Parse.File({ uri: 'www.somewhere.com/image.jpg' })`. This is the same pattern used with the javascript SDK.

Reply:
Oh I see, that is the way to do it.